src/test/resources/features/Purchase.feature

```diff
@@ -8,7 +8,7 @@ Feature: Purchase
     Given I login as "user" with password "password"
     And I post an advertisement with title "Santa" and price "2016"
     And I login as "user2" with password "password"
-    And I post a purchase to advertisement "1"
+    And I post a purchase to advertisement "1" who "user"
```

Comment:
@jesoch, @zurfyx is right, revert Purchase.feature and PurchaseStepDefs.java to previous content. The user associated to the purchase is not determined in the API URL, it is based on the user currently logged in. In this scenarios the user "user"...